code/processes/tailreader.q

```diff
@@ -1,40 +1,71 @@
+determinepartition:{@[value;`.tr.currentpartition;                         /-function to determine the partition value
+  (`date^partitiontype)$(.z.D,.z.d)gmttime]
+  };
+getpartition:@[value;`getpartition;determinepartition[]];
 segmentid: "J"$.proc.params[`segid]
-taildir:hsym `$getenv`KDBTAIL						   /-load in taildir env variables
-.tr.currentpartition:.tr.getpartition;					   /-obtain  partition value
+taildir:hsym `$getenv`KDBTAIL                                              /-load in taildir env variables
+currentpartition:getpartition;                                             /-obtain  partition value
 basedir:(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/"       /-define associated tailer base directory
-wdbdir:`$ basedir,string currentpartition                                      /-define IDB direction
+wdbdir:`$ basedir,string currentpartition                                  /-define IDB direction
+tailertypes:`$"tailer_",last "_" vs string .proc.proctype                  /-define tailer to make connection to 
+.servers.CONNECTIONS:(distinct .servers.CONNECTIONS,.tr.tailertypes) except ` 
+.servers.startup[];
+
+
+\d .ds
+
+getaccess:{[] `location`table xkey update location:.proc.procname,proctype:.proc.proctype from .ds.access};
+
+// function to update the access table in the gateway. Takes the gateway handle as argument
+updategw:{[h]
+
+    newtab:getaccess[];
+    neg[h](`.ds.updateaccess;newtab);
+
+    };
 
 \d .
 endofday:{[pt]
   /- end of day function that will be triggered by EOD Sorter once IDB is copied to HDB
   /-  updates partition and loads in next days partition
   .lg.o[`eod;"End of day message received - ",spt:string pt];
-    currentpartition::pt+1;
-    reload[];
+  .tr.currentpartition:pt+1;
+  reload[];
   }
 
 reload:{
   /- function to define the access table and IDB dir and then reload both tables
   /- reload is triggered by tailer after savedown occurs
-  basedir::(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/";
-  wdbdir::`$ basedir,string currentpartition;
-  accesstabdir::`$ basedir,"access";
-  
- @[.Q.l ;wdbdir;{.lg.e[`load;"Failed to load intradayDB with error: ",x]}];
- @[.Q.l ;accesstabdir;{.lg.e[`load;"Failed to load tailer accesstable with error: ",x]}];
-  load hsym `$basedir,"sym"
+  .tr.basedir:(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/";
+  .tr.wdbdir:`$ .tr.basedir,string .tr.currentpartition;
+  accesstabdir:`$ .tr.basedir,"access";
+  .lg.o[`load;"Loading intradayDB"];
+  @[.Q.l ;.tr.wdbdir;{.lg.e[`load;"Failed to load intradayDB with error: ",x]}];
+  .lg.o[`load;"intradayDB loaded"];
+  .lg.o[`load;"loading accesstable"];
+   /- make a connection to the tailer to get the in-memory access table
+  tailerhandle: first exec w from .servers.getservers[`proctype;.tr.tailertypes;()!();1b;0b];
+  .ds.access:tailerhandle".ds.access";
```

Comment:
N.B. Error handling for this has been implemented on a separate branch/PR (#538).  We will need to ensure the change is there on the final merge.